Latest edits of section 2 (#211)
hassanobeid1994/tr_b_causal_2020: article/sections/_2_graph_importance.tex

We model this travel mode choice problem based on a dataset from \citet{brathwaite_asymmetric}.
This dataset is based on the 2012 California Household Travel Survey, and it
contains approximately 4000 home-based school or work tours made by approximately 3850 individuals in the California Bay area.
The dataset includes eight travel modes. 

> For our illustrative purposes, we focus on the following car-centric modes and will not further present details about other modes:

Reviewer: ```suggestion
For our illustrative purposes, we focus on the following car-centric modes:
```

\begin{itemize}
   \item Drive Alone: The individual uses a private vehicle to make the trip
   \item Shared Ride 2: The individual shares an automobile ride with one more individual
   \item Shared Ride 3+: The individual shares an automobile ride with two or more individuals
\end{itemize}

Readers interested in a more detailed description of the dataset can refer to \citet{brathwaite_asymmetric}.

> For purposes of this exercise, we consider the Multinomial Logit model defined in \citet{brathwaite_asymmetric} as the true outcome generating model.

Reviewer: ```suggestion
For the purposes of this exercise, we treat the multinomial logit model (MNL) defined in \citet{brathwaite_asymmetric} as the true outcome generating model.
```

> The systematic utility equations of the adopted multinomial logit model for the car-centric modes are specified as follows:

Reviewer: ```suggestion
In this model, the systematic utility equations of the car-centric modes defined above are specified as follows:
```

> \begin{equation*}

Reviewer 2: ```suggestion
\begin{equation}
```

   \begin{aligned}

>    \textrm{Utility} \left(\textrm{Drive Alone}\right) &= \beta_{\textrm{travel\_time}} \times \textrm{Travel\_Time} + \beta_{\textrm{cost\_per\_distance\_drive\_alone}} \times \textrm{Cost\_per\_Distance}_{\textrm{da}} \\

Reviewer 2: ```suggestion
   \label{eq:DA}
   \textrm{Utility} \left(\textrm{Drive Alone}\right) &= \beta_{\textrm{travel\_time}} \times \textrm{Travel\_Time} + \beta_{\textrm{cost\_per\_distance\_drive\_alone}} \times \textrm{Cost\_per\_Distance}_{\textrm{da}} \\
```

Reviewer 2: Can you add similar labels ({DA, SR2, SR3}) before each utilities' `\textrm{Utility} \left(\textrm{ ... } \right) &=` line? See example [here](https://stackoverflow.com/a/2599668).

   &\quad + \beta_{\textrm{cost\_per\_distance\_sr3+}} \times \textrm{Cost\_per\_Distance}_{\textrm{sr3+}} + \beta_{\textrm{autos}}  \times \textrm{Number\_of\_Autos} \\
   &\quad + \beta_{\textrm{cross\_bay}} \times \textrm{Cross\_Bay} + \beta_{\textrm{hh\_size}} \times \textrm{Household\_Size} \\
   &\quad + \beta_{\textrm{n\_kids\_hh}} \times \textrm{Number\_of\_kids} \\
   \end{aligned}
\end{equation*}

> For our purposes, we assume that there are no latent variables.

Reviewer: ```suggestion
Note that, since we are treating the model defined in \citet{brathwaite_asymmetric} as the true outcome model, there are no latent variables that we need to account for.
```

Author: I updated your suggestion to drop the citation @hassan-obeid.
```suggestion
Note that, since we consider this model to be the true outcome model, there are no latent variables we need to account for.
```

> The variables were chosen based on the specification of the MNL model outlined by the equations above.

Reviewer: ```suggestion

```

> We can describe the variables as follows:

Reviewer: ```suggestion
Below is a description of the key variables included in the model:
```

  \item Cross-bay trip: a binary variable indicating whether the trip t in tour l for individual i is a cross-bay trip.
\end{itemize}

Figure \ref{fig:IND_GRAPH} illustrates the DAG where all explanatory variables in each utility equation are marginally independent.
This DAG is equivalent to what an analyst assumes when they only update the variables directly impacted by a given policy, without considering the dependencies between the explanatory variables. 
To illustrate the problem with this approach, consider the case where the ``true'' data generating process that reflects the dependencies between the covariates is as shown in Figures \ref{fig:DA_causal_2} through \ref{fig:SR3_causal_2}. 
Under this generative model, intervening on one variable would also result in changes to other variables that are dependent on it.

> The goal of our simulation exercise is to show that ignoring the true generative model of the data can result in arbitrarily biased treatment effects, even if the analyst knows the true outcome model.

Reviewer 2: Assuming you do take the Equation numbering suggestions above,
```suggestion
The goal of our simulation exercise is to show that ignoring the true generative model of the data (Figures \ref{fig:DA_causal_2} through \ref{fig:SR3_causal_2}) can result in arbitrarily biased treatment effects, even if the analyst knows the true outcome model (Equations \ref{eq:DA} through \ref{eq:SR3}).
```

> The goal of our simulation exercise is to show that ignoring the true generative model of the data would result in arbitrarily biased treatment effects, even if the analyst knows the true outcome model.

Reviewer 2: ```suggestion
The goal of our simulation exercise is to show that ignoring the true generative model of the data can result in arbitrarily biased treatment effects, even if the analyst knows the true outcome model.
```

> Figure \ref{fig:IND_GRAPH} illustrates the DAG where all explanatory variables are independent of each other for each utility equation.

Reviewer: ```suggestion
Figure \ref{fig:IND_GRAPH} illustrates the DAG where all explanatory variables in each utility equation are marginally independent.
```

> This DAG represent a scenario where an analyst only modifies variables in their dataset that are directly impacted by the proposed intervention, without considering the different relationships between these variables and other explanatory variables in the dataset.

Reviewer: ```suggestion
This DAG is equivalent to what an analyst assumes when they only update the variables directly impacted by a given policy to study its effect on the outcome, without considering the dependencies between the explanatory variables. 
```

Reviewer 2: @hassan-obeid One suggestion on top of your suggestion: We likely don't need to remind readers that we're studying the effect of a policy on an outcome. This is stated continuously throughout the paper.
```suggestion
This DAG is equivalent to what an analyst assumes when they only update the variables directly impacted by a given policy, without considering the dependencies between the explanatory variables. 
```

> Similarly, Figure \ref{fig:DA_causal_2} through \ref{fig:SR3_causal_2} illustrate the causal graphs with ``dependent'' explanatory variables.
> This DAG represents a scenario where an analyst considers the relationships between all variables and modifies all children nodes of the variable directly impacted by the proposed intervention.
> For our purposes, we assume that these DAGs represent the ``true'' data generating process, where several variables affect each other.
> Each of these graphs is based on the utility function of each mode in the multinomial logit model specified in \citet{brathwaite_asymmetric}.

Reviewer: ```suggestion
To illustrate the problem with this approach, consider the case where the "true" data generating process that reflects the dependencies between the covariates is as shown in Figures \ref{fig:DA_causal_2} through \ref{fig:SR3_causal_2}. 
Under this generative model, intervening on one variable would also result in changes to other variables that are dependent on it.
The goal of our simulation exercise is to show that ignoring the true generative model of the data would result in arbitrarily biased treatment effects, even if the analyst knows the true outcome model.
```

Reviewer 2: lol @hassan-obeid, please use `` '' instead of " " when writing latex.
This has been your achilles heel!

Author: ```suggestion
To illustrate the problem with this approach, consider the case where the ``true'' data generating process that reflects the dependencies between the covariates is as shown in Figures \ref{fig:DA_causal_2} through \ref{fig:SR3_causal_2}. 
Under this generative model, intervening on one variable would also result in changes to other variables that are dependent on it.
The goal of our simulation exercise is to show that ignoring the true generative model of the data would result in arbitrarily biased treatment effects, even if the analyst knows the true outcome model.
```

Author: I fixed it here @timothyb0912

   \label{fig:SR3_causal_2}
\end{figure}

> To achieve this goal, we follow these simulation steps, using the same outcome model:

Reviewer 2: ```suggestion
To achieve this goal, we follow these simulation steps, using the same outcome model defined in Equations \ref{eq:DA} to \ref{eq:SR3}:
```

\begin{itemize}

>    \item Simulate data from the DAGs shown in Figure \ref{fig:IND_GRAPH} through Figure \ref{fig:SR3_causal_2}.

Reviewer 2: ```suggestion
   \item Simulate data from the DAGs shown in Figure \ref{fig:DA_causal_2} through Figure \ref{fig:SR3_causal_2}.
```

Author: This should stay as IND_GRAPH because we simulate data based on the different causal graphs. @timothyb0912

Reviewer 2: @bouzaghrane, here's how I understand the simulation on observable's public notebook:
- From [L565](https://github.com/hassan-obeid/tr_b_causal_2020/blob/master/notebooks/final/7.0-mab-selection-on-observables-final.py#L565) - [L619](https://github.com/hassan-obeid/tr_b_causal_2020/blob/master/notebooks/final/7.0-mab-selection-on-observables-final.py#L619) you simulate data once as `long_sim_data`,
using the `SIMULATE_NODES_WIDE` script,
defined [here](https://github.com/hassan-obeid/tr_b_causal_2020/blob/master/notebooks/final/7.0-mab-selection-on-observables-final.py#L77) to reference [simworkflow.py](https://github.com/hassan-obeid/tr_b_causal_2020/blob/master/src/causal2020/observables/simworkflow.py),
which makes simulations based on `fig:DA_causal2`, hence the [regressions](https://github.com/hassan-obeid/tr_b_causal_2020/blob/master/src/causal2020/observables/simworkflow.py#L24).
- `long_sim_data` is then copied to make `long_sim_data_naive` and `long_sim_data_causal` in [L647](https://github.com/hassan-obeid/tr_b_causal_2020/blob/master/notebooks/final/7.0-mab-selection-on-observables-final.py#L647).
These two dataframes are then further perturbed according to our do-operator / intervention for each graph.

I believe that what I've written above and seen in the code
matches the description of "simulating from `fig:DA_causal2`", not from `fig:IND_GRAPH`.
Let me know if and why (i.e. with links to code supporting why) you believe otherwise!

Reviewer 2: @bouzaghrane Gentle ping on this.

Author: You are right. My apologies, I was just combining all of the simulation effort into one step in my mind.

Reviewer 2: 🙏🏽 No worries, thanks for making the change!

   \item Predict the probabilities of choosing car-centric modes before and after modifying the data to emulate the proposed intervention under each of the two scenarios above, and compute the differences in mode choice probabilities.
\end{itemize}

Readers interested in exploring the details of our simulation exercise can refer to our GitHub repository \citep{brathwaite_etal_2020}.

We then plot histograms of the computed differences between the average probability of an individual
in our sample choosing a car centric mode before and after implementing a policy or intervention
aimed at reducing travel distance.
Recall that these differences are plotted under the assumptions that the outcome model is the same in both scenarios, and the only difference between the two scenarios is the set of variables assumed to be affected by the proposed intervention.

> This difference in the set of variables affected by the proposed intervention is a result of the two different causal graphs representing each scenario (Figures \ref{fig:IND_GRAPH}, \ref{fig:DA_causal_2}, \ref{fig:SR2_causal_2}, and \ref{fig:SR3_causal_2}).

Reviewer 2: ```suggestion
This difference in the set of variables affected by the proposed intervention is a result of the two different causal graphs representing each scenario (Figures \ref{fig:IND_GRAPH} - \ref{fig:SR3_causal_2}).
```

   \label{fig:SR3_causal_2}
\end{figure}

> Using the same outcome model, we:
> \begin{itemize}
>    \item Simulate data from the DAGs shown in Figure \ref{fig:IND_GRAPH} through Figure \ref{fig:SR3_causal_2}
>    \item Modify the travel distance variable in all graphs to emulate a company's decision to move its employees closer to campus, and modify all children nodes of the impacted variable (where appropriate) based on the structure of each DAG.
>    \item Predict the probabilities of choosing car-centric modes before and after modifying the data to emulate the proposed intervention and compute the differences in mode choice probabilities based on the different DAGs
> \end{item}

Reviewer: ```suggestion
To show this, we run the following simulation, using the same outcome model:
\begin{itemize}
   \item We simulate data from the DAGs shown in Figure \ref{fig:IND_GRAPH} through Figure \ref{fig:SR3_causal_2}, representing the true data generating model.
   \item In one scenario, we only modify the travel distance variable in all graphs to emulate a company's decision to move its employees closer to campus. This is similar to assuming the DAG in Figure \ref{fig:IND_GRAPH} as the true data generating model.
   \item In the other scenario, we modify the travel distance variable in all graphs, as well as all the variables that are dependent on it based on the DAGs in Figure \ref{fig:IND_GRAPH} through Figure \ref{fig:SR3_causal_2}. This case is meant to reflect the "correct" approach needed to quantify the effects of the companies policy. 
   \item We predict the probabilities of choosing car-centric modes before and after modifying the data to emulate the proposed intervention under each of the two scenarios above, and compute the differences in mode choice probabilities.
\end{item}
```

Reviewer 2: @hassan-obeid, what do you mean by "To show this"? Looking at the lines before 124, there is no clear "this" that I identify.
Lines 96 - 122 are just the images of causal graphs.
Lines 94 and earlier are describing the data.
It seems line 124 is the first line where we begin describing the simulation.
I don't think it's intuitive / appropriate to refer back to something (e.g. "this") about the simulation yet.

Also, can we keep the instruction-style description of our simulation?
I.e., by beginning each bullet with a verb that immediately tells readers what we did. 
In @bouzaghrane's original version, this was done with "Simulate; Modify; Predict".
I think it'd be easy enough to do the same thing here.

Lastly, can we keep each bullet point to only a single idea / action?
This basically means no ", and" in the sentence.
I would split the second bullet point at ", and modify."
I'd split the third bullet point at "and compute".

Reviewer: @timothyb0912 the "This" refers to line 93, ".. the goal of this simulation" part. I think "To achieve this" might be better actually, but either way, a transitional statement is needed, rather than directly starting with the steps of the simulation. 

I'm confused by your second comment. I don't have ", and" in my suggestions. I don't think the original three bullets convey the idea we're doing correctly (and I'm talking about the core idea, not the semantics of how we communicate it), so I'm not sure there is a point in reviewing that. 

I see your preference for instruction style, but disagree that it's the way to go here. Try re-writing what I wrote with instruction style -- it reads weird to me. Besides, the goal is not to give instructions here, the goal is to describe what we did to illustrate our point, so I don't necessarily see why it should be instruction-style.

Reviewer 2: @hassan-obeid ah I see our confusion now.

In @bouzaghrane's [file](https://github.com/hassan-obeid/tr_b_causal_2020/blob/abdb28bb0e8d5ba21f2da74986d1cab5b4bc0c5f/article/sections/_2_graph_importance.tex), lines 93 - 94 say:
> For our purposes, we assume that these DAGs represent the ``true'' data generating process, where several variables affect each other.
Each of these graphs is based on the utility function of each mode in the multinomial logit model specified in \citet{brathwaite_asymmetric}.

Your [comment](https://github.com/hassan-obeid/tr_b_causal_2020/pull/211#discussion_r550759730) for line 93 says ".. the goal of this simulation".
We were looking at two different line 93s.
Besides this, we're in total agreement about the need for transitional statements.

", and" is in the original bullet points.
Sorry for the confusion caused by including the last paragraph in my earlier comment.
Because it responds directly to @bouzaghrane's writing,
I should have put it in its own comment instead of placing it above.

Author: I am somewhere in the middle of both of you on this. I say that in line 124 we replace it by "To achieve this goal, we follow these simulation steps, using the same outcome model:" and keep the original "Simulate, modify, predict" structure in the bullets with the second bullet broken down like you did.

@hassan-obeid I also suggest dropping the "representing the true data generating process" from the first bullet since it was already mentioned in your suggestion in line 91.

Author: I made some more edits to your suggestions @hassan-obeid. Mainly 1) dropping the "representing the true data generating prorcess" from the first bullet, 2) maintaining a similar structure between bullet 2 and 3 and replacing IND_GRAPH by DA_causal_2, and 3) maintaining the simulate, modify, predict structure.

```suggestion
To achieve this goal, we follow these simulation steps, using the same outcome model:
\begin{itemize}
   \item Simulate data from the DAGs shown in Figure \ref{fig:IND_GRAPH} through Figure \ref{fig:SR3_causal_2}.
   \item In one scenario, modify the travel distance variable in all utility graphs based on the DAG in Figure \ref{fig:IND_GRAPH}. This case assumes that this DAG represents the true data generating model.
   \item In the other scenario, modify the travel distance variable in all utility graphs, as well as all the variables that are dependent on it based on the DAGs in Figure \ref{fig:DA_causal_2} through Figure \ref{fig:SR3_causal_2}. This case is meant to reflect the ``correct'' approach needed to quantify the effects of the company's policy. 
   \item Predict the probabilities of choosing car-centric modes before and after modifying the data to emulate the proposed intervention under each of the two scenarios above, and compute the differences in mode choice probabilities.
\end{itemize}
```

Reviewer: @bouzaghrane I think that the way you rephrased bullet 2 de-emphasizes the main point -- it's not wrong, just the emphasis is not where we want it. My preference is to keep it the way I wrote it, as it emphasizes the fact that we _**only**_ modified travel distance. The way you wrote it, it's left for the reader to figure out that we only modified travel distance, and we're assuming that they'll look at the IND_GRAPH figure and realize that. 
Besides, I find it a bit off to say "Modify the travel distance variable based on a DAG". _Modifying_ a variable is independent of a DAG. That's why I wrote "modify travel distance **_only_**", and "this assumes the IND_GRAPH as the true DAG". The **_only_** part is where the DAG gets involved. 

cc @timothyb0912 for visibility.

Author: @hassan-obeid Hmm I see. I implemented the change in here. I also think that the last sentence in the third bullet is somewhat incorrect. Bullet 3 shows the correct approach for finding out what the data would look like under the true data generating process, not necessarily what the effect of the proposed policy is. I think we're all on the same page and it's just somewhat of a semantics issue here.

@timothyb0912 I'm done with all changes except this one. I will commit this version but leave the comment open for visibility and reference in your final review.

```suggestion
To achieve this goal, we follow these simulation steps, using the same outcome model:
\begin{itemize}
   \item Simulate data from the DAGs shown in Figure \ref{fig:IND_GRAPH} through Figure \ref{fig:SR3_causal_2}.
   \item In one scenario, only modify the travel distance variable in all graphs to emulate a company's decision to move its employees closer to campus. This is similar to assuming the DAG in Figure \ref{fig:IND_GRAPH} as the true data generating model.
   \item In the other scenario, modify the travel distance variable in all utility graphs, as well as all the variables that are dependent on it based on the DAGs in Figure \ref{fig:DA_causal_2} through Figure \ref{fig:SR3_causal_2}. This case is meant to reflect the ``correct'' approach needed to quantify the effects of the company's policy. 
   \item Predict the probabilities of choosing car-centric modes before and after modifying the data to emulate the proposed intervention under each of the two scenarios above, and compute the differences in mode choice probabilities.
\end{itemize}
```

Author: @hassan-obeid let me know if this change I suggested is fair game.

Readers interested in exploring the details of our simulation exercise can refer to our GitHub repository \citep{brathwaite_etal_2020}.

We then plot histograms of the computed differences between the average probability of an individual
in our sample choosing a car centric mode before and after implementing a policy or intervention
aimed at reducing travel distance.

> These differences are plotted under the following assumptions: that the outcome model is the same in both scenarios and that the only difference between the two scenarios is the set of variables affected by the proposed intervention.

Reviewer: ```suggestion
Recall that these differences are plotted under the following assumptions: the outcome model is the same in both scenarios, and the only difference between the two scenarios is the set of variables assumed to be affected by the proposed intervention.
```

Author: More edits to your suggestion @hassan-obeid 

```suggestion
Recall that these differences are plotted under the assumptions that the outcome model is the same in both scenarios, and the only difference between the two scenarios is the set of variables assumed to be affected by the proposed intervention.
```

> This difference in the set of variables affected by the proposed intervention is a result of the two different causal graphs representing each scenario (Figures \ref{fig:IND_GRAPH}, \ref{fig:DA_causal_2}, \ref{fig:SR2_causal_2}, and \ref{fig:SR3_causal_2}).

Reviewer 2: Assuming there's no reason for using SR3_causal_2, I think you meant the following:
```suggestion
This difference in the set of variables affected by the proposed intervention is a result of the two different causal graphs representing each scenario (Figures \ref{fig:IND_GRAPH}, \ref{fig:DA_causal_2}, \ref{fig:SR2_causal_2}, and \ref{fig:DA_causal_2}).
```
